Matrix.js

```diff
@@ -0,0 +1,83 @@
+// file to hold your base Matrix data structure
+// Feel free to use a class you've already written in the past
+
+
+class Matrix {
+    constructor(row, col) {
+        this.row = row
+        this.col = col
+        this.matrix = []
+        this.num = 1
+    }
+    generateMatrix() {
+        for (let r = 0; r < this.row; r++) {
+            this.matrix.push([])
+            for (let c = 0; c < this.col; c++) {
+                this.matrix[r].push(`.`)
+
+            }
+            
+        }
+        this.matrix[0][0] = this.num
+        this.matrix[this.row - 1][this.col - 1] = this.num + 1
+
+      const randomC = Math.floor(Math.random() * (this.col*this.row -2))
+                for (let i = 0; i < randomC; i++){
+                    const randomCol = Math.floor(Math.random() * this.col)
+                    const randomRow = Math.floor(Math.random() * this.row)
+                    const Random = this.matrix[randomRow][randomCol]
+                    if(Random == ".")
+                    {this.alter(randomRow, randomCol, "c")}
+                    
+                }
+                for (let i = 0; i < randomC; i++){
+                    const randomCol = Math.floor(Math.random() * this.col)
+                    const randomRow = Math.floor(Math.random() * this.row)
+                    const Random = this.matrix[randomRow][randomCol]
+                    if(Random == ".")
+                    {this.alter(randomRow, randomCol, "B")}
```

Comment:
there is a repetition here -> lines 25-30 = lines 33-38, except of one parameter ("B" instead of "c") - think how could you do that in a more proper way